burr/core/application.py

```diff
@@ -980,10 +1000,19 @@ def sequence_id(self) -> Optional[int]:
 
         :return:
         """
-        return self._state.get(SEQUENCE_ID, 0)
+        return self._sequence_id
 
     def _increment_sequence_id(self):
-        self._state = self._state.update(**{SEQUENCE_ID: self.sequence_id + 1})
+        self._sequence_id = self._sequence_id + 1
+        self._state = self._state.update(**{SEQUENCE_ID: self.sequence_id})
+
+    @property
+    def id(self) -> str:
+        return self._id
+
+    @property
+    def partition_key(self) -> str:
```

Comment:
Let's add docstrings for these properties and make sure they're part of the docs.